alacritty/src/event.rs

```diff
@@ -1570,24 +1553,33 @@ impl Processor {
                         *control_flow = ControlFlow::Exit;
                     }
                 },
-                // Process all pending events.
-                WinitEvent::RedrawEventsCleared => {
-                    // Check for pending frame callbacks on Wayland.
-                    #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
-                    if let Some(wayland_event_queue) = self.wayland_event_queue.as_mut() {
-                        wayland_event_queue
-                            .dispatch_pending(&mut (), |_, _, _| {})
-                            .expect("failed to dispatch wayland event queue");
-                    }
+                WinitEvent::RedrawRequested(window_id) => {
+                    // Try to handle the events if have end up having some.
+                    let window_context = match self.windows.get_mut(&window_id) {
+                        Some(window_context) => window_context,
+                        None => return,
+                    };
 
+                    window_context.handle_event(
+                        event_loop,
+                        &proxy,
+                        &mut clipboard,
+                        &mut scheduler,
+                        WinitEvent::RedrawRequested(window_id),
+                    );
+
+                    window_context.draw(&mut scheduler);
+                },
+                // Process all pending events.
+                WinitEvent::AboutToWait => {
```

Comment:
What is "about to wait" supposed to mean. It implies it's not waiting yet, so it's still busy?

Reply:
It implies that event loop is about go back to waiting for events.